specs-go/chain/chainid.go

```diff
@@ -0,0 +1,46 @@
+package chain
+
+import "github.com/docker/distribution/digest"
+
+// ChainID takes a slice of digests and returns the ChainID corresponding to
+// the last entry. Typically, these are a list of layer DiffIDs, with the
```

Comment:
> My did you quote yourself as the canonical definition of ChainID?

I quoted myself as the clearest (to me ;) description of ChainID being a property of an array (which you point out is not blindingly obvious in the current spec wording, #482.

> Actually, a list of digests has a ChainID for each entry in the array. This function returns the last entry.

The [spec definition][1] is for ChainID, singular, not a ChainIDs array.  And the ChainID being returned here corresponds to the whole slice, not just the last entry.  Yes, the returned ChainID is the same digest as the last entry in the mutated slice from your `ChainIDs` function, but I think that's an implementation detail that's too peripheral for the opening sentence of this comment.

[1]: https://github.com/opencontainers/image-spec/blob/v1.0.0-rc3/config.md#layer-chainid